sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcSerializer.scala

```diff
@@ -27,12 +27,19 @@ import org.apache.spark.sql.catalyst.util._
 import org.apache.spark.sql.types._
 
 /**
- * A serializer to serialize Spark rows to ORC structs.
+ * A serializer to serialize Spark rows to ORC structs. Multiple calls return the same [[OrcStruct]]
+ * object. If needed, caller should copy before making a new call.
  */
 class OrcSerializer(dataSchema: StructType) {
 
-  private val result = createOrcValue(dataSchema).asInstanceOf[OrcStruct]
-  private val converters = dataSchema.map(_.dataType).map(newConverter(_)).toArray
+  // TODO: repeating steps here. Can pass as parameter to newConverter, but only done once at the
```

Comment:
When you use TODO, could you file a JIRA ID and make this as an IDed TODO like `TODO(SPARK-XXXXX)`, please?

Reply:
Apologies for the delay. Created https://issues.apache.org/jira/browse/SPARK-32732 and added to TODO comment.